migrations/create-patient-contacts.js

```diff
@@ -0,0 +1,177 @@
+      keys: registrationIdsToConsider,
+      include_docs: true
+    }, function(err, results) {
+      if (err) {
+        return callback(err);
+      }
+
+      var uniqueValidRegistrations = _.chain(results.rows)
+        .pluck('doc')
+        .filter(function(registration) {
+          // Registrations require a patient_id to indicate they are the type to
+          // have a patient contact created for them
+          return registration.patient_id;
+        })
+        .uniq(false, function(registration) {
+          // And we only need one for each patient.
+          return registration.patient_id;
+        })
+        .value();
+
+      if (!uniqueValidRegistrations.length) {
+        console.log('no new patient registrations in this batch');
+        return callback();
+      } else {
+        process.stdout.write(uniqueValidRegistrations.length + ' new patient registrations.. ');
+      }
+
+      process.stdout.write('Getting parents.. ');
+
+      var contactPhoneNumbers = _.chain(uniqueValidRegistrations)
+        .pluck('from')
+        .uniq()
+        .map(function(ph) {
+          return [ph];
+        })
+        .value();
+
+      db.medic.view('medic-client', 'people_by_phone', {
+        keys: contactPhoneNumbers,
+        include_docs: true
+      }, function(err, results) {
+        if (err) {
+          return callback(err);
+        }
+
+        var contactForPhoneNumber = _.chain(results.rows)
+          .pluck('doc')
+          .uniq()
+          .reduce(function(memo, doc) {
+            memo[doc.phone] = doc;
+            return memo;
+          }, {})
+          .value();
+
+        var patientPersons = uniqueValidRegistrations.map(function(registration) {
+          var contact = contactForPhoneNumber[registration.from];
+          // create a new patient with this patient_id
+          var patient = {
+            // TODO: Marc is working out the full list of these alternates
+            name: registration.fields.patient_name ||
+                  registration.fields.full_name,
+            parent: contact && contact.parent,
+            reported_date: registration.reported_date,
+            type: 'person',
+            patient_id: registration.patient_id
+          };
+          // include the DOB if it was generated on report
+          if (registration.birth_date) {
+            patient.date_of_birth = registration.birth_date;
+          }
+          return patient;
+        });
+
+        process.stdout.write('Storing ' + patientPersons.length + ' new patient contacts.. ');
+
+        db.medic.bulk({docs: patientPersons}, function(err, results) {
+          if (err) {
+            return callback(err);
+          }
+
+          var errors = results.filter(function(result) {
+            return !result.ok;
+          });
+
+          if (errors.length) {
+            return callback(new Error('Bulk create errors: ' + JSON.stringify(errors)));
+          }
+
+          console.log('DONE');
```

Comment:
I'll change it to `batch done`. It's concluding the one line of output per batch in the log